software/bcftools/tabix/main.nf

```diff
@@ -0,0 +1,33 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BCFTOOLS_TABIX {
+    tag "$meta.id"
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.11" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bcftools:1.11--h7c999a4_0"
+    } else {
+        container "quay.io/biocontainers/bcftools:1.11--h7c999a4_0"
+    }
+
+    input:
+    tuple val(meta), path(gz_genome_position_file)
```

Comment:
```suggestion
    tuple val(meta), path(vcf)
```

Reply:
Wouldn't be misleading to call the channel `vcf` since the module can have as input several file formats apart from `vcf` (`bed`, `gff` and `sam`) ?

Reply:
I think its fine. You are right in that it could be various types of input but the most common use case here would be `vcf`? Maybe we go with that for simplicity and wait until we have figured out what to do for these generic instances? The `vcf` variable won't affect the way the module is run because it can be called anything.